src/NCMenuBar.cc

```diff
@@ -193,13 +187,29 @@ NCMenuBar::wHandleInput(wint_t key)
 
     switch (key) {
 	case KEY_LEFT:
-	    select_previous_menu();
+	{
+	    MenuIterator previous = previousMenu();
+
+	    if ( previous == _menus.end() )
+		previous = lastMenu();
+
+	    selectMenu( previous );
 	    wRedraw();
+
 	    break;
+	}
 	case KEY_RIGHT:
```

Comment:
Please separate the individual cases with a newline as per our C++ coding conventions.